homeassistant/components/sensor/dublin_bus_transport.py

```diff
@@ -0,0 +1,193 @@
+            next_up = "None"
+            if len(self._times) > 1:
+                next_up = self._times[1][ATTR_ROUTE] + " in "
+                next_up += self._times[1][ATTR_DUE_IN]
+
+            return {
+                ATTR_DUE_IN: self._times[0][ATTR_DUE_IN],
+                ATTR_DUE_AT: self._times[0][ATTR_DUE_AT],
+                ATTR_STOP_ID: self._stop,
+                ATTR_ROUTE: self._times[0][ATTR_ROUTE],
+                ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+                ATTR_NEXT_UP: next_up
+            }
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON
+
+    def update(self):
+        """Get the latest data from opendata.ch and update the states."""
+        self.data.update()
+        self._times = self.data.info
+        try:
+            self._state = self._times[0][ATTR_DUE_IN]
+        except TypeError:
+            pass
+
+
+class PublicTransportData(object):
+    """The Class for handling the data retrieval."""
+
+    def __init__(self, stop, route):
+        """Initialize the data object."""
+        self.stop = stop
+        self.route = route
+        self.info = [{ATTR_DUE_AT: 'n/a',
+                      ATTR_ROUTE: self.route,
+                      ATTR_DUE_IN: 'n/a'}]
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest data from opendata.ch."""
+        params = {}
+        params['stopid'] = self.stop
+
+        if len(self.route) > 0:
+            params['routeid'] = self.route
+
+        params['maxresults'] = 2
+        params['format'] = 'json'
+
+        response = requests.get(
+            _RESOURCE,
+            params,
+            timeout=10)
+
+        if response.status_code != 200:
+            self.info = [{ATTR_DUE_AT: 'n/a',
+                          ATTR_ROUTE: self.route,
+                          ATTR_DUE_IN: 'n/a'}]
+            return
+
+        result = response.json()
+
+        if str(result['errorcode']) != '0':
+            self.info = [{ATTR_DUE_AT: 'n/a',
+                          ATTR_ROUTE: self.route,
+                          ATTR_DUE_IN: 'n/a'}]
+            return
+
+        try:
+            self.info = [
+                {ATTR_DUE_AT: item['departuredatetime'],
+                 ATTR_ROUTE: item['route'],
+                 ATTR_DUE_IN: time_in_minutes(
+                     dt_util.as_local(
+                         dt_util.utc_from_timestamp(
+                             dt_util.as_timestamp(
+                                 datetime.strptime(item['departuredatetime'],
+                                                   "%d/%m/%Y %H:%M:%S")))) -
+                     dt_util.as_local(dt_util.utcnow()))}
+                for item in result['results']
+            ]
+
+        except KeyError:
```

Comment:
Instead of catching KeyErrors, is it possible to use `item.get(key, somedefault)`?